sites/hashdev/src/components/Link.tsx

```diff
@@ -10,10 +10,42 @@ import { UrlObject } from "url";
 import { FRONTEND_URL } from "../config";
 import { Button } from "./Button";
 
-export const isHrefExternal = (href: string | UrlObject) =>
-  typeof href === "string" &&
-  (href === "/discord" || !/^(mailto:|#|\/|https:\/\/hash\.dev)/.test(href)) &&
-  !href.startsWith(FRONTEND_URL);
+// List of domains that are considered internal (i.e. should not be opened in a new tab)
+const internalDomains = ["hash.dev", "hash.ai", "blockprotocol.org"];
+
+/**
+ * @param {string | UrlObject} href
+ * @returns {boolean} whether or not the provided href URL is external
+ */
+export const isHrefExternal = (href: string | UrlObject) => {
+  if (typeof href !== "string") {
+    /** @todo: handle NextJs UrlObjects, currently they're always considered external */
+    return false;
+  }
+
+  const isRelativeUrl = href.startsWith("/");
+
+  if (isRelativeUrl) {
+    return href === "/discord";
```

Comment:
@nathggns can I get rid of this? I think it was copied over from the BP site, and doesn't serve a purpose on the hash.dev site because it doesn't have a redirect at `/discord` (unless we're planning to add that?)

Reply:
addressed in https://github.com/hashintel/hash/pull/574/commits/8ea3ac9a81d971896d8fdfe5de14d273bd3b571c